src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONTransportAction.java

```diff
@@ -27,26 +31,62 @@
  */
 public class UploadGeoJSONTransportAction extends HandledTransportAction<UploadGeoJSONRequest, AcknowledgedResponse> {
 
+    private final ClusterService clusterService;
+    private final Client client;
+
     @Inject
-    public UploadGeoJSONTransportAction(TransportService transportService, ActionFilters actionFilters) {
+    public UploadGeoJSONTransportAction(
+        ClusterService clusterService,
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client
+    ) {
         super(UploadGeoJSONAction.NAME, transportService, actionFilters, UploadGeoJSONRequest::new);
+        this.clusterService = clusterService;
+        this.client = client;
     }
 
     @Override
     protected void doExecute(Task task, UploadGeoJSONRequest request, ActionListener<AcknowledgedResponse> actionListener) {
+        UploadGeoJSONRequestContent content = getContent(request, actionListener);
+        if (content == null) {
+            return;
```

Comment:
that is handled inside getContent

Reply:
Whats the advantage of passing the actionListener to a synchronous function? It seems to me to that it would be more readable to keep async failure handling outside of sync function calls.

Reply:
Ack